[CI] SearchableSnapshotEnableAllocationDeciderIntegTests testAllocationDisabled failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/iesmgw5two4y6/tests/:x-pack:plugin:searchable-snapshots:internalClusterTest/org.elasticsearch.xpack.searchablesnapshots.allocation.SearchableSnapshotEnableAllocationDeciderIntegTests/testAllocationDisabled

**Reproduction line:**
`./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.allocation.SearchableSnapshotEnableAllocationDeciderIntegTests.testAllocationDisabled" -Dtests.seed=B37144B9F5C71BEC -Dtests.locale=de-AT -Dtests.timezone=America/Kentucky/Monticello -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.searchablesnapshots.allocation.SearchableSnapshotEnableAllocationDeciderIntegTests&tests.test=testAllocationDisabled

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=875, name=elasticsearch[node_t0][generic][T#6], state=RUNNABLE, group=TGRP-SearchableSnapshotEnableAllocationDeciderIntegTests]

  at __randomizedtesting.SeedInfo.seed([B37144B9F5C71BEC:5311FC9EB41E488A]:0)

  Caused by: java.lang.AssertionError: no previous metadata found for [wvbksxreci/gtSvVp42R9CmXLKQ-Sz8pQ]

    at __randomizedtesting.SeedInfo.seed([B37144B9F5C71BEC]:0)
    at org.elasticsearch.xpack.searchablesnapshots.cache.blob.BlobStoreCacheMaintenanceService$MaintenanceTask.doRun(BlobStoreCacheMaintenanceService.java:146)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

```

--- a/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java
+++ b/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/blob/BlobStoreCacheMaintenanceService.java
@@ -143,7 +143,8 @@ public class BlobStoreCacheMaintenanceService implements ClusterStateListener {
 
             for (Index deletedIndex : event.indicesDeleted()) {
                 final IndexMetadata indexMetadata = event.previousState().metadata().index(deletedIndex);
-                assert indexMetadata != null : "no previous metadata found for " + deletedIndex;
+                assert indexMetadata != null || state.metadata().indexGraveyard().containsIndex(deletedIndex)
+                    : "no previous metadata found for " + deletedIndex;
                 if (indexMetadata != null) {
                     final Settings indexSetting = indexMetadata.getSettings();
                     if (SearchableSnapshotsSettings.isSearchableSnapshotStore(indexSetting)) {

